src/IssueViz/Editor/SelectedIssueTagging/Adornment/IssueLocationAdornmentTagger.cs

```diff
@@ -18,20 +18,46 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Editor;
 using Microsoft.VisualStudio.Text.Tagging;
+using SonarLint.VisualStudio.IssueVisualization.Models;
 
 namespace SonarLint.VisualStudio.IssueVisualization.Editor.SelectedIssueTagging.Adornment
 {
     internal sealed class IssueLocationAdornmentTagger : FilteringTaggerBase<ISelectedIssueLocationTag, IntraTextAdornmentTag>
     {
         private readonly IWpfTextView wpfView;
+        private readonly ICachingAdornmentFactory adornmentCache;
+        private readonly ITagAggregator<ISelectedIssueLocationTag> tagAggregator;
 
         public IssueLocationAdornmentTagger(ITagAggregator<ISelectedIssueLocationTag> tagAggregator, IWpfTextView textView)
-            : base(tagAggregator, textView)
+            : this(tagAggregator, textView, new CachingAdornmentFactory(textView))
         {
-            wpfView = textView;
+        }
+
+        internal /* for testing */ IssueLocationAdornmentTagger(ITagAggregator<ISelectedIssueLocationTag> tagAggregator,
```

Comment:
Why not have the TaggerProvider create the Factory and pass it down? It's not a problem to expose that the AdornmentTagger uses a cache.
In many places of the code we end up with the main class receiving parameters that it doesn't need, just so it could pass it down to the classes that it creates. I think it creates unnecessary complexity and results with the main class knowing too much about the inner class' dependencies.
I think it would be better to have just a public ctor and have the TaggerProvider do the orchestrating.

Reply:
Adding the parameter to the public constructor implies that the caller has a choice about the caching strategy to use, which isn't what is intended. This is pattern we've used in other places where an interface has been solely to enable testing of an internal implementation, which is the case here.
Normally when a caller is responsible for passing in arguments it's because they are shared instances/external dependencies, not internal internal implementation types.